relayer/chain/ethereum/listener.go

```diff
@@ -107,23 +110,47 @@ func (li *Listener) pollEventsAndHeaders(ctx context.Context) error {
 				li.messages <- *msg
 			}
 		case gethheader := <-headers:
```

Comment:
I feel this code needs to be made fork aware for subsequent submission of blocks. It should have some sort of queue where we connect the blocks and submit them one by one, reordering any chains of blocks when they are coming in. 

As this code currently stands, it might not pick up forks if the more difficult fork is a bit slower.

Reply:
@philipstanislaus the onchain code is fork aware already - though I think you're right in that we want to make the relayer eventually fork aware too. I don't think we should address this in the current PR though.